src/cert.rs

```diff
@@ -117,7 +134,19 @@ impl CertContext {
                         (*chain_ptr).rgpElement,
                         (*chain_ptr).cElement as usize,
                     );
+
+                    let mut first = true;
                     for element in elements {
+                        if first {
```

Comment:
Instead of the additional mutable flag you can also use `.enumerate()`:
```rust
for (i, element) in elements.iter().enumerate() {
   if i > 0 {
       ...
   }
}
```

Reply:
done